data-prepper-api/src/main/java/org/opensearch/dataprepper/model/buffer/Buffer.java

```diff
@@ -138,4 +139,24 @@ default boolean isWrittenOffHeapOnly() {
      */
     default void shutdown() {
     }
+
+    /**
+     * Sets default failure pipeline of a source
+
+     * @param failurePipeline failure pipeline
+     * @since 2.12
+     */
+    default void setFailurePipeline(final FailurePipeline failurePipeline) {
+    }
+
+    /**
+     * Returns default failure pipeline of a source
+
+     * @return FailurePipeline returns failure pipeline
+     * @since 2.12
+     */
+    default FailurePipeline getFailurePipeline() {
```

Comment:
I don't think we need these in the interface. Each `Buffer` can handle `setFailurePipeline` as needed.